prime-angular-frontend/src/app/modules/adjudication/shared/components/search-ha-form/search-ha-form.component.ts

```diff
@@ -0,0 +1,153 @@
+    this.assignToMe = new EventEmitter<boolean>();
+    this.refresh = new EventEmitter<void>();
+  }
+
+  public get textSearch(): FormControl {
+    return this.form.get('textSearch') as FormControl;
+  }
+
+  public get siteStatusCode(): FormControl {
+    return this.form.get('siteStatusCode') as FormControl;
+  }
+
+  public get vendorCode(): FormControl {
+    return this.form.get('vendorCode') as FormControl;
+  }
+
+  public get careTypeCode(): FormControl {
+    return this.form.get('careTypeCode') as FormControl;
+  }
+
+  public get assignToMeCkbx(): FormControl {
+    return this.form.get('assignToMe') as FormControl;
+  }
+
+  public onRefresh() {
+    this.refresh.emit();
+  }
+
+  public ngOnInit() {
+    this.createFormInstance();
+    this.initForm();
+  }
+
+  private createFormInstance() {
+    this.form = this.fb.group({
+      textSearch: [null, []],
+      assignToMe: [{ value: false }, []],
+      siteStatusCode: [{ value: '' }, []],
+      vendorCode: [{ value: '' }, []],
+      careTypeCode: [{ value: '' }, []],
+    });
+  }
+
+  private initForm() {
+
+    this.textSearch.valueChanges
+      .pipe(debounceTime(500))
+      // Passing `null` removes the query parameter from the URL
+      .subscribe((search: string) => {
+        this.localStorage.set(this.textSearchKey, search || '');
+        this.search.emit(search || null);
+      });
+
+    this.siteStatusCode.valueChanges
+      .pipe(debounceTime(500))
+      // Passing `null` removes the query parameter from the URL
+      .subscribe((siteStatus: number) => {
+        this.localStorage.set(this.siteStatusCodeKey, siteStatus?.toString());
+        this.siteStatus.emit(siteStatus || null);
+      });
+
+    this.vendorCode.valueChanges
+      .pipe(debounceTime(500))
+      // Passing `null` removes the query parameter from the URL
+      .subscribe((vendor: number) => {
+        this.localStorage.set(this.vendorCodeKey, vendor?.toString());
+        this.vendor.emit(vendor || null);
+      });
+
+    this.careTypeCode.valueChanges
+      .pipe(debounceTime(500))
+      // Passing `null` removes the query parameter from the URL
+      .subscribe((careType: number) => {
+        this.localStorage.set(this.careTypeCodeKey, careType?.toString());
+        this.careType.emit(careType || null);
+      });
+
+    this.assignToMeCkbx.valueChanges
+      .pipe(debounceTime(500))
+      // Passing `null` removes the query parameter from the URL
+      .subscribe((assignToMe: boolean) => {
+        this.localStorage.set(this.assignToMeCodeKey, assignToMe.toString());
+        this.assignToMe.emit(assignToMe);
+      });
+
+    this.form.patchValue({
+      textSearch: this.localStorage.get(this.textSearchKey),
+      siteStatusCode: this.localStorage.getInteger(this.siteStatusCodeKey) || null,
+      vendorCode: this.localStorage.getInteger(this.vendorCodeKey) || null,
+      careTypeCode: this.localStorage.getInteger(this.careTypeCodeKey) || null,
```

Comment:
Care Type doesn't seem to be saved/restored properly?  
![image](https://github.com/bcgov/moh-prime/assets/7586665/eeb8d5a6-5db0-4f32-b0a7-c769f97a1313)
versus
![image](https://github.com/bcgov/moh-prime/assets/7586665/6e5de6a0-f07d-44d6-abec-adc6e32013d1)

Reply:
Good catch. It is fixed.